### DISCUSSION TITLE
Thank you so much! @ClassicOldSong! You're awesome!

### DISCUSSION THREAD

**Discussion by shantanub08** [ClassicOldSong/moonlight-android#69] [General]
I read the readme.md for Apollo and Artemis and got some context behind the birth of both and why it was necessary.

Recently got my keyboard cover for Samsung Galaxy Tab S9 Ultra, thinking I can finally have a laptop-like experience with the addition of Remote Desktop approach. Tailscale and moonlight was already setup on the tablet along with sunshine on my desktop.

My first experience was so awful. I couldn't scroll with 2 finger gestures on the trackpad, right-click was not getting registered on server-end, and few other bad experiences which started to make me regret my decision of buying this expensive keyboard. [Googled for solution](https://github.com/moonlight-stream/moonlight-android/issues/1259#issuecomment-2265126676), and then came across this amazing fork. The installation of Artemis was so seamless! 99% of the default settings are already great for start working right away! I was amazed by how many additional features are provided here as compared to OG Moonlight. As if Artemis was actually made with all the practicalities in mind, and for actually using it for practical applications.

I also understand why no real-time chat support exists for either application. But I wanted to express my gratitude one way or another. So sorry if such discussions aren't supposed to be created on GitHub. Even if it's removed, I'd be happy to know @ClassicOldSong at least read it once. You're amazing and thanks for putting all of this hard work into making Artemis! (I haven't even tried Apollo but I think I'm going to have an even greater experience!!!) Also happy to know they apologized for what they did to you.

I wrote this whole message through Artemis itself (will switch sunshine with Apollo soon)

**Comment by ClassicOldSong** [comment]
Thank you for your appreciation! That's warm harted!

**Reply by hellosad59** [reply]
Hello, please read my comments 👍

**Reply by hellosad59** [reply]
Hello in @ClassicOldSong your recent client update with added "Warp Drive" very positively affected the gameplay in games, the feeling that it became much smoother and better. The phone became much less loaded, if anything I have a snapdragon 732g, so it is not high-performance phone, and the average one is like that. So thank you very much for the work you've done in this project, although it's written that the new function should be used in high-performance devices, but It also has a good effect on the weak.

**Reply by hellosad59** [reply]
And I compared the previous functions with this new one (with recording, my broadcast from the computer was lagging then) the new function showed itself well, in that everything goes much smoother even with the recording.

**Reply by hellosad59** [reply]
And in other functions there were freezes or lags with drops

**Reply by ClassicOldSong** [reply]
Interesting, glad to know it helps also on lower end devices! Still need to be tested on more device to be sure it actually works...

**Reply by hellosad59** [reply]
I hope this function will be further developed, my game literally works longer (the phone uses less power and therefore consumes less energy) and there is less heating 🥰🥰🥰

**Reply by hellosad59** [reply]
And thanks again for such a good development of this program, because in winter it is cold to sit at the computer, but from the phone you can play cool and with low latency

**Reply by DeriXrace** [reply]
now I use warp2 and I will say right away that everything has become better, on the PC the load is negligibly less (this is for me) comparing in 1 game warp1 and warp2 in smoothness it is clear that the 2nd version is better. And the joke is that the sound has a much greater delay than the game itself and input control 😂😂😂. So you need to somehow configure the sound for low latency

**Reply by DeriXrace** [reply]
I'm that hellosad59, I don't remember which gmail I used to log in. I just changed the firmware 2 days ago)

**Reply by hellosad59** [reply]
I'm a fool, I made a mistake with 1 letter in gmail 😂😂😂😂

**Reply by hellosad59** [reply]
Oh, I'm talking about the fact that the input lag is so low (And displaying the game on the screen) that the sound has a big delay

**Comment by hellosad59** [comment]
![Screenshot_20241224-004651_Artemis](https://github.com/user-attachments/assets/4d9aecac-104c-40f3-8baf-cae0e466c8cd)
The cool thing is that the load is literally The cool thing is that the load is literally below 10 percent.

**Reply by hellosad59** [reply]
The screenshot shows that it is 30% but these are just jumps, most often it is 15%
